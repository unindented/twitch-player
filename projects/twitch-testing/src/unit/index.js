import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
