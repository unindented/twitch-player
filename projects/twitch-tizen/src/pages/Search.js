import Heading from "@twitch-player/ui/dist/components/Heading";
import React from "react";

const Search = () => <Heading level="2">Search</Heading>;

export default Search;
