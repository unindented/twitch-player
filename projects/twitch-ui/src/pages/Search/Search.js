import React, { memo } from "react";
import { View } from "react-native";
import Heading from "../../components/Heading";

const Search = memo(({ testID = "search-page" }) => (
  <View testID={testID}>
    <Heading level="2">Search</Heading>
  </View>
));

export default Search;
