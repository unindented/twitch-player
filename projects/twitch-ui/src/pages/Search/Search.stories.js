import { storiesOf } from "@storybook/react";
import React from "react";
import Search from ".";

storiesOf("Pages|Search", module).add("default", () => <Search />);
