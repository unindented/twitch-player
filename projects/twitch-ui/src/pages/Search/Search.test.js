import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import Search from "./Search";

describe("Search", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Search />);
  });

  it("renders the page", () => {
    expect(instance.getByTestId("search-page")).toMatchSnapshot();
  });
});
