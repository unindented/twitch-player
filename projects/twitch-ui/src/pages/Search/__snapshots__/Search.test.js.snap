// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders the page 1`] = `
<DocumentFragment>
  <div
    class="css-view-1dbjc4n"
  >
    <h2
      aria-level="2"
      class="css-reset-4rbku5 css-text-76zvg2 r-fontWeight-vw2c0b"
      data-testid="heading"
      dir="auto"
      role="heading"
      style="color: rgb(50, 47, 55); font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Helvetica Neue',sans-serif; font-size: 16px; line-height: 1.5; margin: 20px 30px 5px 30px;"
    >
      Search
    </h2>
  </div>
</DocumentFragment>
`;
